lib/dependencies/HarmonyImportSpecifierDependency.js

```diff
@@ -320,23 +320,23 @@ HarmonyImportSpecifierDependency.Template = class HarmonyImportSpecifierDependen
 
 		let [rangeStart, rangeEnd] = dep.range;
 		if (trimmedIds.length !== ids.length) {
-			// The array returned from dep.idRangeStarts is right-aligned with the array returned from dep.getIds.
+			// The array returned from dep.idRanges is right-aligned with the array returned from dep.getIds.
 			// Meaning, the two arrays may not always have the same number of elements, but the last element of
-			// dep.idRangeStarts corresponds to [the starting range position of] the last element of dep.getIds.
-			// Use this to find the correct range end position based on the number of ids that were trimmed.
+			// dep.idRanges corresponds to [the expression fragment to the left of] the last element of dep.getIds.
+			// Use this to find the correct replacement range based on the number of ids that were trimmed.
 			const idx =
-				dep.idRangeStarts === undefined
+				dep.idRanges === undefined
 					? -1 /* trigger failure case below */
-					: dep.idRangeStarts.length + (trimmedIds.length - ids.length);
-			if (idx < 0 || idx >= dep.idRangeStarts.length) {
+					: dep.idRanges.length + (trimmedIds.length - ids.length);
+			if (idx < 0 || idx >= dep.idRanges.length) {
 				// cspell:ignore minifiers
 				// Should not happen but we can't throw an error here because of backward compatibility with
 				// external plugins in wp5.  Instead, we just disable trimming for now.  This may break some minifiers.
 				trimmedIds = ids;
 				// TODO webpack 6 remove the "trimmedIds = ids" above and uncomment the following line instead.
 				// throw new Error("Missing range starts data for id replacement trimming.");
 			} else {
-				rangeEnd = dep.idRangeStarts[idx];
+				[rangeStart, rangeEnd] = dep.idRanges[idx];
```

Comment:
This is Important Part 2 of the change.  Use the entire subrange and not just the end.